Binaries/DafnyRuntime.go

```diff
@@ -1358,28 +1362,44 @@ func (mset MultiSet) EqualsGeneric(other interface{}) bool {
 // IsSubsetOf returns whether one multiset has a subset of the elements of the
 // other, with lesser or equal multiplicities.
 func (mset MultiSet) IsSubsetOf(mset2 MultiSet) bool {
-  return mset.isRelated(mset2, func(x, y int) bool { return x <= y })
+  return mset.isRelated(
+    mset2,
+    func(x, y int) bool { return x <= y },
+    func(isProper bool) bool { return true },
+  )
 }
 
 // IsProperSubsetOf returns whether one multiset has a proper subset of the
 // elements of the other, with strictly lesser multiplicities.
 func (mset MultiSet) IsProperSubsetOf(mset2 MultiSet) bool {
-  return mset.isRelated(mset2, func(x, y int) bool { return x < y })
+  return mset.isRelated(
+    mset2,
+    func(x, y int) bool { return x <= y },
+    func(isProper bool) bool { return isProper },
+  )
 }
 
-func (mset MultiSet) isRelated(mset2 MultiSet, r func(int, int) bool) bool {
+func (mset MultiSet) isRelated(
+  mset2 MultiSet,
+  r func(int, int) bool,
+  properCheck func(bool) bool,
+) bool {
   if !r(len(mset.elts), len(mset2.elts)) {
     return false
   }
-
+  isProper := len(mset.elts) < len(mset2.elts)
   for _, e := range mset.elts {
     m := mset2.Multiplicity(e.value)
-    if !r(e.count.Cmp(m), 0) {
+    cmp := e.count.Cmp(m)
```

Comment:
`.Cmp` wasn't used on line 1387, so is it needed here? Would it not be easier to call `r(e.count, m)` and to use `e.count != m` on line 1397?

Reply:
It is needed here because the arguments are bigints, not ints.